_tsinfermodule.c

```diff
@@ -1535,6 +1536,12 @@ AncestorMatcher_get_total_memory(AncestorMatcher *self, void *closure)
 {
     PyObject *ret = NULL;
 
```

Comment:
This will potentially break some tests. Better to do something like this:
```c
unsigned long val;

...

#if defined(TSI_NO_ATOMICS)
     val = (unsigned long) PY_SSIZE_T_MAX);
#else    
     val = ancestor_matcher_get_total_memory(self->ancestor_matcher));
endif;
ret = Py_BuildValue("k", val);

```

Always keep control flow as simple as possible in C.